### PR TITLE
Parallel pin source fetching

### LINKED ISSUE #4315: pin-depends pinning speed
I recently tried to `opam install` a local package with every dependency pinned via the `pin-depends` field in its opam file and the `opam install` is significantly slowed down by the initial pinning process.

I tried the 2.1 alpha hoping it might have been improved by recent changes but it seems to have no effect.

Do you have any idea why this takes so long and/or if this can be easily improved? I suspect that cloning the repositories is what takes up most of the time but I also imagine running those in parallel could improve the situation.

### PR DESCRIPTION
**Pull request by rjbou**
Fix #4315
Using the same example, it took 4m32 (with download-only option), and now with parallel and reuse of source dir, it takes only 1m32.
It also applies to `pin scan` arguments, and url pin.
cc/ @NathanReb

### CODE REVIEW

File: src/client/opamPinCommand.ml [hunk 1]
```diff
@@ -366,6 +366,43 @@ let default_version st name =
...
+    List.fold_left (fun (err,ok) result ->
+        let name, url, subpath, result = result in
+        match result with
+        | Not_available _ -> (* clean dir ? *)
+          (name, url, subpath)::err, ok
+        | _ -> err, (url, subpath)::ok)
+      ([],[]) fetched
+  in
+  if errored = []
+  || OpamConsole.confirm "Could not retrieve%s\nContinue?"
```
Comment (AltGr) [inline src/client/opamPinCommand.ml:396]: We may want to be more clear about the implications of saying "yes" here ?
Also, be careful that `yes` is the default and may be forced with `--yes` misled people.

**Review by AltGr — COMMENTED** [review]
Seems all good to me, just one `confirm` call that may not be clear enough to the user.

File: src/client/opamPinCommand.ml (hunk 1 above)
Comment (rjbou) [reply src/client/opamPinCommand.ml:396]: Maybe adding something like "You may reach an inconsistent state because of those not pinned packages"?

Comment (rjbou) [reply src/client/opamPinCommand.ml]: updated

**Pull request merged**